AppCenter/AppCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -38,32 +38,14 @@ - (instancetype)initWithSchema:(MSDBSchema *)schema version:(NSUInteger)version
     }
   });
   if ((self = [super init])) {
-    BOOL newDatabase = ![MSUtility fileExistsForPathComponent:filename];
-    _dbFileURL = [MSUtility createFileAtPathComponent:filename withData:nil atomically:NO forceOverwrite:NO];
-    _maxSizeInBytes = kMSDefaultDatabaseSizeInBytes;
-
-    int result;
-    sqlite3 *db = [MSDBStorage openDatabaseAtFileURL:self.dbFileURL withResult:&result];
-    if (db) {
-      _pageSize = [MSDBStorage getPageSizeInOpenedDatabase:db];
-      NSUInteger databaseVersion = [MSDBStorage versionInOpenedDatabase:db];
-
-      // Create table
-      if (schema) {
-        [MSDBStorage createTablesWithSchema:schema inOpenedDatabase:db];
-      }
-      if (newDatabase) {
-        MSLogInfo([MSAppCenter logTag], @"Created \"%@\" database with %lu version.", filename, (unsigned long)version);
-        [self customizeDatabase:db];
-      } else if (databaseVersion < version) {
-        MSLogInfo([MSAppCenter logTag], @"Migrating \"%@\" database from version %lu to %lu.", filename, (unsigned long)databaseVersion,
-                  (unsigned long)version);
-        [self migrateDatabase:db fromVersion:databaseVersion];
-      }
-      [MSDBStorage enableAutoVacuumInOpenedDatabase:db];
-      [MSDBStorage setVersion:version inOpenedDatabase:db];
-      sqlite3_close(db);
-    };
+    int result = [self initializeDatabaseWithSchema:schema version:version filename:filename];
+    if (result == SQLITE_CORRUPT || result == SQLITE_NOTADB) {
+      [self dropDatabase];
+      result = [self initializeDatabaseWithSchema:schema version:version filename:filename];
+    }
+    if (result != SQLITE_OK) {
+      MSLogError([MSAppCenter logTag], @"Failed to initialize database.");
```

Comment:
I'd like to start including the SQLITE result code in the error log

Reply:
It is already printed in the previous line